sbt-crossproject/src/main/scala/sbtcrossproject/CrossType.scala

```diff
@@ -69,9 +87,29 @@ object CrossType {
     def sharedSrcDir(projectBase: File, conf: String): Option[File] =
       Some(projectBase.getParentFile / "shared" / "src" / conf / "scala")
 
+    override def partiallySharedSrcDir(projectBase: File,
+                                       platforms: Seq[Platform],
+                                       conf: String): Option[File] =
+      if (platforms.isEmpty) {
+        None
+      } else {
+        val dir = platforms.map(_.identifier).mkString("-")
+        Some(projectBase.getParentFile / dir / "src" / conf / "scala")
+      }
+
     override def sharedResourcesDir(projectBase: File,
                                     conf: String): Option[File] =
       Some(projectBase.getParentFile / "shared" / "src" / conf / "resources")
+
+    override def partiallySharedResourcesDir(projectBase: File,
+                                             platforms: Seq[Platform],
+                                             conf: String): Option[File] =
+      if (platforms.isEmpty) {
+        None
```

Comment:
Every implementation will always have to do that. Instead, we should make it a pre-condition of `partiallyShared{Src,Resources}Dir` that `platforms.nonEmpty`. This is a bit more work for the (unique) caller of these methods, and less work for everyone who would like to implement `CrossType`.

Reply:
This comment is still relevant, I believe.

Reply:
Woops, I thought I got this. Must have been another one 🤔